lib/SPIRV/OCLUtil.cpp

```diff
@@ -143,10 +143,21 @@ unsigned getOCLVersion(Module *M) {
   NamedMDNode *NamedMD = M->getNamedMetadata(kSPIR2MD::OCLVer);
   if (!NamedMD)
     return 0;
-  assert (NamedMD->getNumOperands() == 1 && "Invalid SPIR");
-  MDNode *MD = NamedMD->getOperand(0);
-  unsigned Major = getMDOperandAsInt(MD, 0);
-  unsigned Minor = getMDOperandAsInt(MD, 1);
+  assert (NamedMD->getNumOperands() > 0 && "Invalid SPIR");
```

Comment:
Ok. I would prefer to have original implementation of getOCLVersion and have another function that will fix duplication of metadata. Do you think we can do that?

New implementation is doing double work by checking metadata which are supposed to be constant.
Even for the first time call it's better to be initialized with the first operand of the named metadata and after that check in the loop that all other operands have the same value.

MDNode *MD = NamedMD->getOperand(0);
unsigned Major = getMDOperandAsInt(MD, 0);
unsigned Minor = getMDOperandAsInt(MD, 1);
for (unsigned I = 1, E = NamedMD->getNumOperands(); I != E; ++I) {
    MDNode *MD = NamedMD->getOperand(I);
    if (Major != getMDOperandAsInt(MD, 0)
        || Minor != getMDOperandAsInt(MD, 1)) {
      report_fatal_error("OCL version mismatch");
    }
  }